htdocs/core/modules/commande/doc/doc_generic_order_odt.modules.php

```diff
@@ -305,8 +305,10 @@ public function write_file($object, $outputlangs, $srctemplatepath, $hidedetails
 				$contactobject = null;
 				if (!empty($usecontact))
 				{
-					if ($usecontact && ($object->contact->fk_soc != $object->thirdparty->id && (!isset($conf->global->MAIN_USE_COMPANY_NAME_OF_CONTACT) || !empty($conf->global->MAIN_USE_COMPANY_NAME_OF_CONTACT)))) {
-						$socobject = $object->contact;
+					if ($usecontact && ($object->contact->fk_soc != $object->thirdparty->id && (!empty($conf->global->MAIN_USE_COMPANY_NAME_OF_CONTACT)))) {
```

Comment:
Th isset was to avoid warning with php8 when param is not set

Reply:
I see. But it makes no sense to me. Because if the param is not set then !isset becomes true

If the param is set, then !empty becomes true.

So the expression is always true

Reply:
... so probably the solution would be

(isset($conf->global->MAIN_USE_COMPANY_NAME_OF_CONTACT) && !empty($conf->global->MAIN_USE_COMPANY_NAME_OF_CONTACT))

Reply:
Yes, correct code should be 
`(isset($conf->global->MAIN_USE_COMPANY_NAME_OF_CONTACT) && !empty($conf->global->MAIN_USE_COMPANY_NAME_OF_CONTACT))`

Reply:
PR is updated